### PR TITLE
Call wakeup after pair for BLE proxy compatibility

### PR DESCRIPTION
**Pull request by pedropombeiro**
## Summary

Narrow upstreamable subset extracted from the broader BLE-proxy work in our
fork: call `wakeup()` after `pair()` instead of bundled inside
`IdasenDesk.connect()`.

## Scope

This PR intentionally does **not** include the full set of changes currently
running in `pedropombeiro/idasen-ha`.

The deployed fork also carries additional BLE-proxy hardening that is out of
scope here, including:
- `bleak-retry-connector`-based connection establishment
- managed external BLE client handoff / wrapper logic
- BLEDevice refresh on reconnect and stale-connection cleanup
- proxy-specific move-state fixes

This PR focuses only on the minimal sequencing fix that appears suitable for
upstreaming on its own.

## Why

`IdasenDesk.connect()` calls `_client.connect()` then `wakeup()` immediately.
Through ESPHome Bluetooth proxies, pairing/authentication may not have completed
by the time `wakeup()` writes to GATT characteristics, causing repeated
`Insufficient authentication` (`status=5`) errors.

This PR splits the connect call so that:
1. `_client.connect()` establishes the BLE link
2. `pair()` completes authentication
3. `wakeup()` runs only after auth succeeds

## Before and after

Before (ESPHome Bluetooth proxy logs):
```text
[W][bluetooth_proxy.connection:348]: [0] [ED:FA:0A:FF:94:FB] Error writing char/descriptor for handle 0x E, status=5
[W][bluetooth_proxy.connection:348]: [0] [ED:FA:0A:FF:94:FB] Error writing char/descriptor for handle 0x E, status=5
[I][esp32_ble_client:570]: [0] [ED:FA:0A:FF:94:FB] auth complete addr: ED:FA:0A:FF:94:FB
```

After:
```text
[I][esp32_ble_client:343]: [0] [ED:FA:0A:FF:94:FB] Connection open
[I][esp32_ble_client:570]: [0] [ED:FA:0A:FF:94:FB] auth complete addr: ED:FA:0A:FF:94:FB
```

No `status=5` errors, auth completes cleanly before wakeup writes.

## Changes

- `connection_manager.py`: replace `self._idasen_desk.connect()` with
  `self._idasen_desk._client.connect()`, add `wakeup()` after `pair()` with
  its own error handling block
- Updated test fixture to mock `_client.connect` and `wakeup` separately
- Added regression tests for wakeup ordering, wakeup failure, and edge cases
- Coverage: 98% total, `connection_manager.py` at 100%

Related to home-assistant/core#155912.

### CODE REVIEW

File: tests/test_connection_management.py [hunk 1]
```diff
@@ -151,12 +156,134 @@ async def test_disconnect_on_pair_failure(
...
+
+    async def wakeup_side_effect():
+        steps.append("wakeup")
+
+    mock_idasen_desk.pair.side_effect = pair_side_effect
+    mock_idasen_desk.wakeup.side_effect = wakeup_side_effect
+
+    await desk.connect(FAKE_BLE_DEVICE)
+
+    assert steps == ["pair", "wakeup"]
```
Comment (abmantis) [inline tests/test_connection_management.py]:
```suggestion
    desk = Desk(Mock(), False)
    await desk.connect(FAKE_BLE_DEVICE)
    mock_idasen_desk.assert_has_calls([call.pair(), call.wakeup()])
```

File: idasen_ha/connection_manager.py [hunk 2]
```diff
@@ -69,7 +69,7 @@ async def _connect(self, retry: bool) -> None:
         try:
             try:
                 _LOGGER.info("Connecting...")
-                await self._idasen_desk.connect()
+                await self._idasen_desk._client.connect()  # noqa: SLF001
```
Comment (abmantis) [inline idasen_ha/connection_manager.py:74]: lets add a comment here too, explaining why we are using an internal

File: tests/test_connection_management.py [hunk 3]
```diff
@@ -151,12 +156,134 @@ async def test_disconnect_on_pair_failure(
...
+
+    with pytest.raises(BleakError):
+        await desk.connect(FAKE_BLE_DEVICE, retry=False)
+
+    mock_idasen_desk.pair.assert_called_once()
+    mock_idasen_desk.disconnect.assert_called()
+    assert update_callback.call_count == 0
+
+
+async def test_pair_failure_non_auth_dbus_error(mock_idasen_desk: MagicMock):
```
Comment (abmantis) [inline tests/test_connection_management.py]: isn't this already tested in `test_disconnect_on_pair_failure`?

File: tests/test_connection_management.py [hunk 4]
```diff
@@ -151,12 +156,134 @@ async def test_disconnect_on_pair_failure(
...
+
+    mock_idasen_desk.pair.side_effect = BleakDBusError("org.bluez.Error.Failed", "")
+    with pytest.raises(BleakDBusError):
+        await desk.connect(FAKE_BLE_DEVICE, retry=False)
+
+    mock_idasen_desk.disconnect.assert_called()
+
+
+@mock.patch("idasen_ha.connection_manager.asyncio.sleep")
+async def test_wakeup_failure_with_retry(
```
Comment (abmantis) [inline tests/test_connection_management.py]: can't we add a new param to test_connect_exception_retry_success instad of this test?

File: tests/test_connection_management.py [hunk 5]
```diff
@@ -151,12 +156,134 @@ async def test_disconnect_on_pair_failure(
...
+    desk = Desk(Mock(), False)
+    mock_idasen_desk.wakeup.side_effect = BleakError()
+    await desk.connect(FAKE_BLE_DEVICE)
+
+    await retry_future
+    assert desk.is_connected
+
+
+@mock.patch("idasen_ha.connection_manager.asyncio.sleep")
+async def test_schedule_reconnect_skipped_when_already_pending(
```
Comment (abmantis) [inline tests/test_connection_management.py]: duplicate of test_connect_called_while_retry_pending?

File: tests/test_connection_management.py [hunk 6]
```diff
@@ -151,12 +156,134 @@ async def test_disconnect_on_pair_failure(
...
+
+    await desk.connect(FAKE_BLE_DEVICE)
+    await desk.connect(FAKE_BLE_DEVICE)
+
+    await retry_future
+    assert sleep_mock.call_count == 2
+
+
+@mock.patch("idasen_ha.connection_manager.asyncio.sleep")
+async def test_reconnect_aborted_when_not_keep_connected(
```
Comment (abmantis) [inline tests/test_connection_management.py]: I am not sure I understand this test. the test name/comment and the implementation do not match

**Review by abmantis — CHANGES_REQUESTED** [review]

File: tests/test_connection_management.py (hunk 1 above)
Comment (pedropombeiro) [reply tests/test_connection_management.py]: Done — simplified to use `assert_has_calls`.

File: idasen_ha/connection_manager.py (hunk 2 above)
Comment (pedropombeiro) [reply idasen_ha/connection_manager.py:74]: Added a comment explaining why we call `_client.connect()` directly instead of `IdasenDesk.connect()`.

File: tests/test_connection_management.py (hunk 3 above)
Comment (pedropombeiro) [reply tests/test_connection_management.py]: You're right — I folded it into the existing `test_disconnect_on_pair_failure` parametrized test as a third case with `BleakDBusError("org.bluez.Error.Failed", "")`.

File: tests/test_connection_management.py (hunk 4 above)
Comment (pedropombeiro) [reply tests/test_connection_management.py]: Done — added `"wakeup"` to the `fail_call_name` parametrize list in both `test_connect_exception_retry_with_disconnect` and `test_connect_exception_retry_success`.

File: tests/test_connection_management.py (hunk 5 above)
Comment (pedropombeiro) [reply tests/test_connection_management.py]: Agreed — removed.

File: tests/test_connection_management.py (hunk 6 above)
Comment (pedropombeiro) [reply tests/test_connection_management.py]: Agreed, it was confusing — removed.

**Review by abmantis — APPROVED** [review]
Thanks!

**Comment by abmantis** [comment]
I've opened a PR to bump the dependency on HA: https://github.com/home-assistant/core/pull/166645

I have tested this with my desk (stock idasen controller) and it is so much faster to connect! Nice work!
